tests/bud.bats

```diff
@@ -6486,6 +6486,15 @@ _EOF
   done
 }
 
+@test "build must reset platform for stages if needed" {
+  otherarch="arm64"
+  # just make sure that other arch is not equivalent to host arch
+  if [[ "$otherarch" == "$myarch" ]]; then
+    otherarch="amd64"
+  fi
+  run_buildah build $WITH_POLICY_JSON --build-arg FOREIGNARCH=$otherarch -f $BUDFILES/multiarch/Containerfile.reset-platform $BUDFILES/multiarch
```

Comment:
This should also test a case where the the TARGETPLATFORM build-arg is set, so that we don't regress on that.

Reply:
I added a test which sets TARGETPLATFORM using `--platform`, should that cover what you meant by this comment ?

Reply:
I mean literally setting `--build-arg TARGETPLATFORM=`, which has been known to be done instead of using the `--platform` flag.  This had to be fixed before in https://github.com/containers/buildah/pull/5731.